sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        address::{AccountAddress, Bech32Address},
+        output::{unlock_condition::AddressUnlockCondition, DelegationId, DelegationOutputBuilder},
+    },
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet},
+};
+
+/// Params for `create_delegation_output()`
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateDelegationParams {
+    /// Bech32 encoded address which will control the delegation.
+    /// By default, the ed25519 wallet address will be used.
+    // TODO: https://github.com/iotaledger/iota-sdk/issues/1888
+    pub address: Option<Bech32Address>,
+    /// The amount to delegate.
+    pub delegated_amount: u64,
+    /// The Account Address of the validator to which this output will delegate.
+    pub validator_address: AccountAddress,
+}
+
+/// The result of a transaction to create a delegation
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateDelegationTransaction {
+    pub delegation_id: DelegationId,
+    pub transaction: TransactionWithMetadata,
+}
+
+/// The result of preparing a transaction to create a delegation
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PreparedCreateDelegationTransaction {
+    pub delegation_id: DelegationId,
+    pub transaction: PreparedTransactionData,
+}
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Creates a delegation output.
+    /// ```ignore
+    /// let params = CreateDelegationParams {
+    ///     address: None,
+    ///     delegated_amount: 200,
```

Comment:
Maybe 2_000_000 would be a more realistic value here, otherwise one just gets an error because it doesn't cover the storage deposit

Reply:
fair

Reply:
done